qiskit/visualization/pulse_v2/generators/waveform.py

```diff
@@ -105,10 +105,8 @@ def gen_filled_waveform_stepwise(
             if isinstance(pval, circuit.ParameterExpression):
                 unbound_params.append(pname)
 
-        if hasattr(data.inst.pulse, "pulse_type"):
-            pulse_shape = data.inst.pulse.pulse_type
-        else:
-            pulse_shape = data.inst.pulse.__class__.__name__
+        pulse_data = data.inst.pulse
+        pulse_shape = getattr(pulse_data, "pulse_type", pulse_data.__class__.__name__)
```

Comment:
All the `getattr(pulse_data, "pulse_type", pulse_data.__class__.__name__)` is a bit ugly. Perhaps `Pulse` could set `self.pulse_type = self.__class__.__name__` in its `__init__`?

Reply:
I assumed Waveform here. Perhaps I should write it explicitly.

Reply:
Done in [a506e19](https://github.com/Qiskit/qiskit-terra/pull/8278/commits/a506e19fde1c00a00dd395253af5102a178800c5)